utils/auto_dockerfile_update/update_dockerfiles.py

```diff
@@ -40,13 +40,133 @@ def is_docker_file_outdated(dockerfile: Dict, latest_tag: str, last_updated: str
+                to_replace = True
+        elif "pyproject.toml" in file_path:
+            if full_str_python_version != f"python = \"~{version[0]}.{version[1]}\"":
+                file_content=file_content.replace(full_str_python_version,f"python = \"~{version[0]}.{version[1]}\"")
+                to_replace= True
+        if to_replace:
+            with open(file_path, "w") as f:
+                f.write(file_content)
+            return True
+        return False
+
+def change_pyproject_or_pipfile(file_path, str_version) -> Tuple[bool,str]:
+    """Replace the current python version in the Pipfile or the pyproject.toml.
+    
+        Args:
+            file_path (str): The file path to the Pipfile or the pyproject.toml file.
+            version (List): The updated version.
+            full_str_python_version (str): The older version.
+    """
+    version = str_version.split(".")
+    current_version, full_str_python_version, success_extraction=extract_current_python_version(file_path)
+    if success_extraction:
+        result=replace_python_version(file_path, version, full_str_python_version)
+        return result,current_version
+    else:
+        print(f"[ERROR] can't extract python version form:{file_path}")
+    return False, []
+
+def run_lock(base_path_docker:str,pipfile_or_pyproject_path:str)->bool:
+    """Runs poetry lock --no-update or pipfile lock --keep-outdated.
+    
+        Args:
+            base_path_docker (str): The DockerFile path.
+            pipfile_or_pyproject_path (str): The file path to the Pipfile or the pyproject.toml file.
+    """
+    base_path=base_path_docker.replace("/Dockerfile", "")
+    current_directory = os.getcwd()
+    os.chdir(f"{current_directory}/" + base_path)
+    try:
+        if "Pipfile" in pipfile_or_pyproject_path:
+            # waits for the process to end.
+            completed_process = subprocess.run(["pipenv", "lock", "--keep-outdated"], check=True)
+            if completed_process.returncode != 0 :
+                return False
+        elif "pyproject.toml" in pipfile_or_pyproject_path:
+            completed_process = subprocess.run(["poetry", "lock", "--no-update"], check=True)
+            if completed_process.returncode != 0 :
+                return False
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] Lock failed with error: {e}")
+    except TimeoutError as e:
+        print(f"[ERROR] Got time out error: {e} for {base_path_docker}")
+    except Exception as e:
+        print(f"[ERROR] {e}: for {base_path_docker}")
+    os.chdir(current_directory)
+    return True
+
+def update_python_version_pyproject_or_pipfile(dockerfile: Dict, latest_tag: str)-> None:
+    """
+    Updating dockerfile content with the latest tag
+    Args:
+        dockerfile (Dict): Dockerfile dict.
+        latest_tag (str): latest tag string.
+    Returns:
+        None
+    """
+    update_result = False
+    image_name = dockerfile['image_name']
```

Comment:
image_name is the base image?

Reply:
No, is the name in the directory

Reply:
if this is the case we will only update python. we want to update every image based on python as well

Reply:
I am sorry I my mistake this is the name of the base image- can be "alpine", "debian", "python", "powershell",
"python3-deb", "python3", "powershell-ubuntu", "crypto", "python3-ubi", "btfl-soup" etc...